src/main/java/interface_adapter/logout/LogoutPresenter.java

```diff
@@ -0,0 +1,35 @@
+package interface_adapter.logout;
+
+import interface_adapter.ViewManagerModel;
+import interface_adapter.login.LoginState;
+import interface_adapter.login.LoginViewModel;
+import use_case.logout.LogoutOutputBoundary;
+
+/**
+ * Presenter for the Logout Use Case.
+ */
+public class LogoutPresenter implements LogoutOutputBoundary {
+
+    private final LoginViewModel loginViewModel;
+    private final ViewManagerModel viewManagerModel;
+
+    public LogoutPresenter(LoginViewModel clubLoggedInViewModel, ViewManagerModel viewManagerModel) {
+        this.loginViewModel = clubLoggedInViewModel;
+        this.viewManagerModel = viewManagerModel;
+    }
+
+    @Override
+    public void switchToLoginView() {
+        // Empties the login state and sets it to the login view model
+        final LoginState loginState = loginViewModel.getState();
+        loginState.setIdentifier("");
+        loginState.setPassword("");
+        loginState.setLoginError(null);
+        this.loginViewModel.setState(loginState);
+        loginViewModel.firePropertyChanged();
```

Comment:
would it not be easier to create anew (and so empty) instance of LoginState, and set the LoginViewModel to that state?

Reply:
I wanted to prevent any hard dependencies and creating more unnecessary objects when we can modify the existing one